homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,326 @@
+        return HVAC_MODES_WITHOUT_OFF
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set HVAC mode to comfort(HEAT) or back to normal(AUTO)."""
+        if hvac_mode == HVAC_MODE_AUTO:
+            await self.async_set_preset_mode(PRESET_NONE)
+            self._attr_hvac_mode = hvac_mode
+        elif hvac_mode == HVAC_MODE_HEAT:
+            await self.async_set_preset_mode(PRESET_COMFORT)
+            self._attr_hvac_mode = hvac_mode
+
+        if self._can_turn_off:
+            if hvac_mode == HVAC_MODE_OFF:
+                await self.async_set_preset_mode(PRESET_NONE)
+                self._attr_hvac_mode = hvac_mode
+                await self._nobo.async_update_zone(
+                    self._id, week_profile_id=self._command_off_id
+                )  # Change week profile to OFF
+                _LOGGER.debug(
+                    "Turned off heater %s '%s' by switching to week profile %s",
+                    self._id,
+                    self._attr_name,
+                    self._command_off_id,
+                )
+            else:
+                await self._nobo.async_update_zone(
+                    self._id, week_profile_id=self._command_on_id
+                )  # Change week profile to normal for this zone
+                _LOGGER.debug(
+                    "Turned on heater %s '%s' by switching to week profile %s",
+                    self._id,
+                    self._attr_name,
+                    self._command_on_id,
+                )
+            # When switching between AUTO and OFF an immediate update does not work (the Nobø API seems to answer with old values), but it works if we add a short delay:
+            await asyncio.sleep(0.5)
+        elif hvac_mode == HVAC_MODE_OFF:
+            _LOGGER.warning(
+                "User tried to turn off zone %s '%s', but this is not configured so this should be impossible",
+                self._id,
+                self._attr_name,
+            )
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new zone override."""
+        if self._nobo.zones[self._id][ATTR_OVERRIDE_ALLOWED] != "1":
+            return
+        if preset_mode == PRESET_ECO:
+            mode = nobo.API.OVERRIDE_MODE_ECO
+        elif preset_mode == PRESET_AWAY:
+            mode = nobo.API.OVERRIDE_MODE_AWAY
+        elif preset_mode == PRESET_COMFORT:
+            mode = nobo.API.OVERRIDE_MODE_COMFORT
+        else:  # PRESET_NONE
+            mode = nobo.API.OVERRIDE_MODE_NORMAL
+        await self._nobo.async_create_override(
+            mode,
+            self._override_type,
+            nobo.API.OVERRIDE_TARGET_ZONE,
+            self._id,
+        )
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        low = int(kwargs.get(ATTR_TARGET_TEMP_LOW))
+        high = int(kwargs.get(ATTR_TARGET_TEMP_HIGH))
+        if low > int(self._nobo.zones[self._id][ATTR_TEMP_COMFORT_C]):
```

Comment:
Is this equivalent to something like this?
```
low = max(int(kwargs.get(ATTR_TARGET_TEMP_LOW), int(self._nobo.zones[self._id][ATTR_TEMP_COMFORT_C])
```

Reply:
I must admin I didn't write this method originally, I only ported it to be async. I think the point is that it's not legal to set the low temperature higher than the high temperatur, and the other away around. However, both checks are against the _existing_ high/low temperatures.

Could the entire adjustment be skipped? I.e., is an HVAC component with support flag `SUPPORT_TARGET_TEMPERATURE_RANGE` guaranteed to always receive both `ATTR_TARGET_TEMP_LOW` and `ATTR_TARGET_TEMP_HIGH` , and that `low <= high`?

Reply:
I am not sure: maybe manually test out with a service call and see what happens..

Reply:
Made the method more robust (and readable).